libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -166,18 +170,25 @@ void CompilerUtils::storeInMemoryDynamic(Type const& _type, bool _padToWordBound
 		m_context << Instruction::DUP2 << Instruction::MSTORE;
 		m_context << u256(_padToWordBoundaries ? 32 : 24) << Instruction::ADD;
 	}
-	else
+	else // process all other types
 	{
 		unsigned numBytes = prepareMemoryStore(_type, _padToWordBoundaries);
-		if (numBytes > 0)
-		{
-			solUnimplementedAssert(
-				_type.sizeOnStack() == 1,
-				"Memory store of types with stack size != 1 not implemented."
-			);
-			m_context << Instruction::DUP2 << Instruction::MSTORE;
-			m_context << u256(numBytes) << Instruction::ADD;
-		}
+		solAssert(
+			numBytes > 0,
+			"Memory store of type " + _type.richIdentifier() + " not possible (stack size 0)."
```

Comment:
`prepareMemoryStore` does not return the stack size but rather the size in memory.

Please also add the assert inside `prepareMemoryStore`.

Reply:
> prepareMemoryStore does not return the stack size but rather the size in memory.

ok

> Please also add the assert inside prepareMemoryStore.

This introduces duplication. Better to move the assertion within prepareMemoryStore.

Leaving the following commits to keep it traceable.